### PR TITLE
[libc] Remove unused target conditionals for Apple platforms

### PR DESCRIPTION
**Pull request by ldionne**
It seems that LIBC_TARGET_OS_IS_MACOS and LIBC_TARGET_OS_IS_IPHONE were never actually used in the code, so these definitions can be removed.

I came across these because libc++ now depends on llvm-libc to build (for from_chars), and the unguarded use of TargetConditionals.h broke some of our downstream configurations. There are some platforms for which `__APPLE__` is defined but that don't provide TargetConditionals.h.

If there is a need to keep defining those, the compiler also provides some uglier macro definitions like `__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__` that do not require including any header.

### CODE REVIEW

**Review by michaelrj-google — APPROVED** [review]
LGTM, we can look into other solutions if necessary. For now it looks like `#ifdef __APPLE__` is good enough for what we're currently doing.

**Comment by ldionne** [comment]
What's the policy for merging? I assume this can be merged once CI is green?

**Comment by michaelrj-google** [comment]
Correct. If this patch was urgent it would be okay to merge before the CI finished, but for normal patches we recommend waiting for the CI.

**Pull request merged**
